src/Trakx.Kaiko.ApiClient/ClientRegistrations.cs

```diff
@@ -25,40 +32,27 @@ private static void AddClients(this IServiceCollection services)
                         onRetry: (result, timeSpan, retryCount, context) =>
                         {
                             var logger = Log.Logger.ForContext<AggregatesClient>();
-                            LogFailure(logger, result, timeSpan, retryCount, context);
+                            logger.LogApiFailure(result, timeSpan, retryCount, context);
                         })
-                    .WithPolicyKey("AggregatesClient"));
+                    .WithPolicyKey("Trakx.Kaiko.ApiClient.AggregatesClient"));
```

Comment:
Instead of using strings to name this, shouldn't it be better to use something like, for instance, `nameof(CLASS)`?

Or is this an auto-generated code? If it is, please ignore this comment.

Reply:
Everything in Trakx.Kaiko.ApiClient was inherited and not exactly what my focus is. My focus is the Stream client, which is what we need for the live price stream. I will fix issues with Kaiko.ApiClient, but there's no immediate use of it.

Reply:
This comes from a T4 template. I am weary of using nameof(.) as it only gives us the local name, not the full name e.g. Trakx.Kaiko.ApiClient.x . There is a proposal asking for a fullnameof() method but is still unanswered. https://github.com/dotnet/csharplang/discussions/701

I will submit a fix to use nameof for the client name, but keep the namespace as a string.

Reply:
I take it back: the class names are actually hardcoded in the T4 template as well:
`var clientNames = new [] { "AggregatesClient", "ExchangesClient" };`

It's too much trouble to replace this, I'd need to add Reflection to the T4 template. Considering we are not using the REST Api Client anywhere for now, we can leave this for a next sprint.

Reply:
I can generate the Reflection code in the T4 template and have it execute in runtime, i.e. CLientRegistrations.cs. Working on it.